src/rng.h

```diff
@@ -183,5 +183,43 @@ cata::optional<tripoint> random_point( const tripoint_range<tripoint> &range,
 /// Same as other random_point with a range enclosing all valid points of the map.
 cata::optional<tripoint> random_point( const map &m,
                                        const std::function<bool( const tripoint & )> &predicate );
+/**
+ * Helper class that provides the means to fully tile w x h matrix in (w x h) steps in
+ * pseudo-random stable way. Internally uses w+h extra space.
+ * @tparam w  width of the matrix
+ * @tparam h  height of the matrix
+ */
+template<size_t w, size_t h>
+class pseudo_random_matrix_tiling
+{
+        static_assert( w > 0 && h > 0, "dimensions should be positive" );
+
+    public:
+        pseudo_random_matrix_tiling( uint64_t seed ) {
+            for( size_t i = 0; i < w; ++i ) {
+                w_arr[i] = i;
+            }
+            for( size_t i = 0; i < h; ++i ) {
+                h_arr[i] = i;
+            }
```

Comment:
Could use `std::iota` for simplicity here:
```suggestion
            std::iota( w_arr.begin(), w_arr.end(), 0 );
            std::iota( h_arr.begin(), h_arr.end(), 0 );
```